### DISCUSSION TITLE
Where is the source code for the proxy?

### DISCUSSION THREAD

**Discussion by GQAdonis** [Budibase/budibase#6321] [Help] [answered]
Good morning!  I love your product. I was wondering where the source code is for the `budibase/proxy` image, as I would like to know if there are limitations to what ports the other services could use, etc.

I intend on perhaps extending Budibase with some of my own integrations to other services I have written (e.g., an asset management service I wrote that uses IPFS for storage and allows for metadata configuration).

It looks as if the image is based on Nginx, but that is all I have been able to find.  Any help on this?

**Comment by Rory-Powell** [comment, marked as answer]
Hey @GQAdonis, you're right it's based on NGINX. The base configuration for this is a handlebars template:
https://github.com/Budibase/budibase/blob/v1.0.200/hosting/nginx.prod.conf.hbs

The configuration is generated using this script:
https://github.com/Budibase/budibase/blob/v1.0.200/scripts/proxy/generateProxyConfig.js

Which is invoked as a yarn command from here and used to build an image
https://github.com/Budibase/budibase/blob/v1.0.200/package.json#L59
https://github.com/Budibase/budibase/blob/v1.0.200/hosting/proxy/Dockerfile

**Comment by GQAdonis** [comment]
That is really slick work!

I am really liking your product. I do wish you had official support for
GraphQL, though….

On Wed, Jun 15, 2022 at 4:05 AM Rory Powell ***@***.***>
wrote:

> Hey @GQAdonis <https://github.com/GQAdonis>, you're right it's based on
> NGINX. The base configuration for this is a handlebars template:
>
> https://github.com/Budibase/budibase/blob/v1.0.200/hosting/nginx.prod.conf.hbs
>
> The configuration is generated using this script:
>
> https://github.com/Budibase/budibase/blob/v1.0.200/scripts/proxy/generateProxyConfig.js
>
> Which is invoked as a yarn command from here and used to build an image
> https://github.com/Budibase/budibase/blob/v1.0.200/package.json#L59
> https://github.com/Budibase/budibase/blob/v1.0.200/hosting/proxy/Dockerfile
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Budibase/budibase/discussions/6321#discussioncomment-2954191>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAYU7QUGUHICL342T6FSFTVPGMF7ANCNFSM5YY7ONYQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 


*TRAVIS JAMES*: Chief Technology Officer at TribeMedia.io
<https://www.tribemedia.io>
phone: 214-449-2472
web: www.tribemedia.io
twitter: @GQAdonisCTO <http://twitter.com/GQAdonisCTO>
